### PR TITLE
Remove redundant conditional check for true and false when the boolean is right there. 

### PR DESCRIPTION
**Pull request by ghost**
Basically, in many lines of code, there are tertiary functions that check if a statement is true, and if that statement is true, assign true to a boolean, and false if the statement is false. However, we can directly set the result of the statement to the boolean. It is much more cleaner and efficient that way.

### CODE REVIEW

**Comment by pi1024e** [comment]
@swift-ci please smoke test and merge

File: lib/Migrator/overlay4.json [hunk 1]
```diff
@@ -1342,5 +1342,5 @@
     "RightUsr": "",
     "RightComment": "firstIndex",
     "ModuleName": "Swift"
-  },
+  }
```
Comment (akyrtzi) [inline lib/Migrator/overlay4.json]: Why are there changes in the `json` files?

Comment (pi1024e) [reply lib/Migrator/overlay4.json]: My editor did that because trailing comma. I will remove that. Sorry!

**Review by akyrtzi — APPROVED** [review]

**Comment by pi1024e** [comment]
@akyrtzi can you please initiate the smoke-tests and merge tests?

**Comment by akyrtzi** [comment]
@swift-ci smoke test and merge

**Comment by akyrtzi** [comment]
@swift-ci smoke test and merge

File: lib/SILOptimizer/Transforms/RedundantLoadElimination.cpp [hunk 2]
```diff
@@ -1563,8 +1563,7 @@ bool RLEContext::run() {
     return false;
 
   // Do we run a multi-iteration data flow ?
-  bool Optimistic = Kind == ProcessKind::ProcessMultipleIterations ?
-                        true : false;
+  bool Optimistic = Kind == ProcessKind::ProcessMultipleIterations;
```
Comment (CodaFi) [inline lib/SILOptimizer/Transforms/RedundantLoadElimination.cpp]: Could you mark these `const` and surround them in parens?

File: lib/SILOptimizer/Transforms/DeadStoreElimination.cpp [hunk 3]
```diff
@@ -1201,7 +1201,7 @@ bool DSEContext::run() {
     return false;
 
   // Do we run a pessimistic data flow ?
-  bool Optimistic = Kind == ProcessKind::ProcessOptimistic ? true : false;
+  bool Optimistic = Kind == ProcessKind::ProcessOptimistic;
```
Comment (CodaFi) [inline lib/SILOptimizer/Transforms/DeadStoreElimination.cpp]: Same here

**Comment by pi1024e** [comment]
Did!

**Comment by pi1024e** [comment]
@CodaFi run the test and merge one more time please

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
Same here. You don’t have to rebase every time.

@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
*Oh*, are you rebasing every time the Windows bot fails? 

Unfortunately, that's a self-perpetuating cycle. The Windows bot currently doesn't auto-reschedule itself after you force push, so the first CI invocation after a force-push is bound to fail. Windows is not a hard gate you need to pass (yet) so don't worry about it. If the bot breaks, we'll just revert the PR and you can give it another go later.

**Comment by CodaFi** [comment]
Here, let's get a real run in.

@swift-ci test Windows

**Review by thejohnlima — APPROVED** [review]

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**